### DISCUSSION TITLE
v3 Beta of "serversideup/php" is now available! 🚀

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/docker-php#254] [Announcements]
Hey folks,

Great news! The v3 beta is now available for **serverisdeup/php**!

We're extremely grateful for the support we've received as we worked to refactor our images to give you the best experience for deploying PHP. We invested **250+ hours** to completely reimagine how we can take these images to the next level.

## 🤩 See the new site
* **🌍 [See the Docs site](https://serversideup.net/open-source/docker-php/)** - Get access to our beta latest docs on our website.

## 🚀 New Features
We've been busy overhauling our PHP Docker Images to make them more production-ready and easier to use. Here are some of the new features we've added:

- **Based on official PHP Images** - We're now building an improved developer experience on top of the official PHP Docker images.
- **PHP 8.3 support** - We're now shipping the latest and greatest.
- **Pin to the exact minor version** - Pin your app to the exact minor version of PHP that you want to use. This means you can pin to `8.2.12` instead of `8.2`.
- **Easier start up script customization** - We now have a folder called `/etc/entrypoint.d` that allows you to easily customize your container with scripts. Just put them in numerical order and we'll execute any shell script you want. No S6 Overlay knowledge required.
- **Expanded Laravel Automations** - We added automations to run `config:cache`, `route:cache`, `view:cache`, `event:cache`, `migrate --force --isolated`, and `storage:link`
- **NGINX Unit Support** - We're offering NGINX Unit as a variation as an alternative to PHP-FPM. This allows you to run PHP applications without the need for a webserver like NGINX or Apache to run with PHP-FPM.
- **Available on GitHub Packages** - We're now publishing our images to GitHub Packages. This means you can use our images without needing to authenticate with Docker Hub.

## Summary of our images vs the official PHP images
![image](https://github.com/serversideup/docker-php/assets/3174134/f4f0bd78-a3df-42c0-8fe4-b4831d6da408)

## ⚠️ Breaking changes
This release is a change in the direction for these images, but we're confident these changes are all for the good. Some major changes to know about:
- Ubuntu is no longer used in our base image
- `ppa:ondrej/php` is no longer used because we get PHP directly from PHP.net
- S6 Overlay is only used in `*-fpm-apache` and `*-fpm-nginx` images
- `webuser` is no longer being used
- `SSL_MODE` is now set to `off` by default (HTTP only)
- `AUTORUN_ENABLED` is now set to `false` by default

To learn more about these changes, we put together a migration guide.

[Read the migration guide from V2 to the V3 beta →](https://serversideup.net/open-source/docker-php/docs/guide/migrating-from-v2-to-v3)

If you'd like to learn more why we made all these changes, see the pull request for the v3 release: https://github.com/serversideup/docker-php/pull/207

## ⏬ Getting the beta images
In this release we are pushing images to **DockerHub** and **GitHub Packages**. Feel free to pull images from either repository. They are the exact same.

* **🐳 [View on DockerHub →](https://hub.docker.com/r/serversideup/php/tags?page=1&name=beta-)**
* **📦 [View on GitHub Packages →](https://github.com/serversideup/docker-php/pkgs/container/php)**
* **👨‍💻 [View the source code →](https://github.com/serversideup/docker-php/tree/release/v3.0)**

## 🙋‍♂️ Getting Help
We want you to have the best experience during the beta, so we put together a preview site so you can get access to the latest documentation.

* **🐛 [Known issues](https://github.com/serversideup/docker-php/labels/%F0%9F%91%A8%E2%80%8D%F0%9F%94%AC%20Release%3A%20Pre-release)** - See and track the known issues of the beta.
* **📖 [Read the docs](https://serversideup.net/open-source/docker-php/docs)** - Get access to our latest docs on our preview site.
* **🗣️ [GitHub Discussions](https://github.com/serversideup/docker-php/discussions)** - Ask for help on our GitHub community.
* **⚡️ [Discord](https://serversideup.net/discord)** - Get friendly help from our community members.
* **🤵‍♂️ [Get Professional Help](https://serversideup.net/professional-support)** - Get video + screen-sharing support from the core contributors.

## 🫵 We need YOUR help
The biggest thing we need from the community is testing, feedback, and support of helping each other out. Specifically, could you help us with the items below?

1. [Are you concerned about image sizes?](https://release-v3-0.docker-php.pages.dev/docs/getting-started/contributing#published-beta-images)
2. What was painful in migrating from v2 to v3?
3. [Do the default modules make sense?](https://release-v3-0.docker-php.pages.dev/docs/getting-started/default-configurations)
4. [What user should we use?](https://github.com/serversideup/docker-php/issues/253) Currently it’s www-data.
5. Please performance test against other images and post your results (compare them to other images and our current images)
6. [Chime in on "Help Wanted" issues](https://github.com/serversideup/docker-php/issues?q=is%3Aopen+label%3A%22%F0%9F%99%8F+Help+Wanted%22+label%3A%22%F0%9F%91%A8%E2%80%8D%F0%9F%94%AC+Release%3A+Pre-release%22+)
7. [Help us improve the docs →](https://github.com/serversideup/docker-php/tree/main/docs)

## ❤️ Submitting Feedback and helping us out
We're so grateful for the support we get from the community. Please use the guide above to submit any feedback that you have. We're looking forward to building an awesome product together, and it's not possible 

If you want to help us out:
* **❎ Share us on X (Twitter)** - ([serversideup](https://x.com/serversideup), [danpastori](https://x.com/danpastori), [jaydrogers](https://x.com/jaydrogers))
* **⭐️ [Star us on GitHub](https://github.com/serversideup/docker-php)**
* **📬 [Sign up on our mailing list]( https://serversideup.net/subscribe/)**
* **🥰 [Consider sponsoring our work](https://github.com/sponsors/serversideup)** 
* **⚡️ [Buy our other products](https://serversideup.net/store/)**

**Comment by antoniovj1** [comment]
I successfully migrated a project from serversideup/php:8.2-fpm-nginx to serversideup/php:beta-8.3-fpm-nginx-alpine without encountering any issues, except for the transition from Ubuntu to Alpine. 

The transition was seamless, thanks to the documentation.

The process was smooth, and everything is now running seamlessly.

Thanks!!

**Reply by jaydrogers** [reply]
Thanks for sharing! I am very glad it worked out!! Now that [Spin](https://serversideup.net/open-source/spin/) is done, I am going to start converting our own projects as well.

Looking forward to hitting getting the stable release merged soon 😃

**Reply by antoniovj1** [reply]
Wow! I'm going to check this Spin thing! But before that I'm going to try to figure out how opcache is configured and If is needed to be customised for my Symfony app.

Thanks for your work!

**Reply by jaydrogers** [reply]
Thanks for your compliments!

Opcache is something definitely on my radar that I would like to improve.

Honestly, I have never played around with Opcache before but I understand it's importance. If there's something that you'd like to see improved, open up a discussion and I would love to chat more about it https://github.com/serversideup/docker-php/discussions

**Comment by LorenzoRogai** [comment]
Is there a release date for the non-beta 8.3 version? The version is listed here but it doesn't exists

https://serversideup.net/open-source/docker-php/docs/getting-started/installation

<img width="648" alt="image" src="https://github.com/serversideup/docker-php/assets/1665768/3c5c21cc-d7ee-4a27-86c6-a70127ac14be">

Also it seems that not everything of this list is supported for the v3

https://serversideup.net/open-source/docker-php/docs/reference/environment-variable-specification

PHP_OPCACHE_ENABLE was not being recognized and we had to add "RUN install-php-extensions opcache" inside our Dockerfile

Without opcache Laravel took 500ms/1sec for each request instead of 50/60ms

**Reply by jaydrogers** [reply]
Sorry for the confusion. Anything on the website references the `beta` images only.

I'm running the beta images on some small production sites now. I don't have a date as of now, but I can assure that I am working very hard to get this out ASAP.

This is hopefully the final breaking change in the beta before rolling it out: https://github.com/serversideup/docker-php/pull/311
